packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -245,6 +270,16 @@ export class RecordSet extends Resource implements IRecordSet {
   constructor(scope: Construct, id: string, props: RecordSetProps) {
     super(scope, id);
 
```

Comment:
Great! I've added a check for when they are defined simultaneously.

Reply:
@lpizzinidev 
As you said, specifying setIdentifier in simple routing caused the deployment to fail...

```ts
    new route53.ARecord(this, 'Record', {
      zone: hostedZone,
      recordName: 'www',
      target: route53.RecordTarget.fromIpAddresses('1.2.3.4'),
      setIdentifier: 'uniqueId'
    });
```

```sh
simple-routing-with-setIdentifier | 1:32:02 AM | CREATE_FAILED        | AWS::Route53::RecordSet  | Record (Record83264F3E) Invalid request: Expected exactly one of [Weight, Region, Failover, GeoLocation, MultiValueAnswer, GeoProximityLocation, or CidrRoutingConfig], but found none in Change with [Action=CREATE, Name=www.cdk.dev., Type=A, SetIdentifier=unique-id] (Service: AmazonRoute53; Status Code: 400; Error Code: InvalidInput; Request ID: 47a20d64-62c9-476b-bc37-10b74e80105f; Proxy: null)
```

I have added logic to prevent the specification of setIdentifier during simple routing. Additionally, I have made corresponding modifications to the documentation.